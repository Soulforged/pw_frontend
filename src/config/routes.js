//@flow
import { Dashboard } from "src/dashboard";
import { Users, UserForm, UserDetails } from "src/users";
import { Roles, RoleForm, RoleDetails } from "src/roles";
import { BusinessUnits, BusinessUnitForm, BusinessUnitDetail } from "src/businessUnits";

export default [
  {
    path: "/",
    name: "home",
    title: "Home",
    component: Dashboard,
    exact: true,
    hidden: true
  },
  {
    path: "/dashboard",
    name: "dashboard",
    title: "Dashboard",
    component: Dashboard,
    exact: true
  },
  {
    path: "/users",
    name: "users",
    title: "Users",
    component: Users,
  },
  {
    path: "/users/:id",
    name: "user",
    title: "User details",
    component: UserDetails,
    hidden: true,
    strict: true,
    exact: true
  },
  {
    path: "/edit/users/:id",
    name: "users_edit",
    title: "User edition",
    component: UserForm,
    hidden: true,
    exact: true,
    strict: true
  },
  {
    path: "/new/users",
    name: "users_create",
    title: "New user",
    component: UserForm,
    hidden: true,
    exact: true,
    strict: true
  },
  {
    path: "/roles",
    name: "roles",
    title: "roles",
    component: Roles,
  },
  {
    path: "/roles/:id",
    name: "role",
    title: "Role details",
    component: RoleDetails,
    hidden: true,
    strict: true,
    exact: true
  },
  {
    path: "/edit/roles/:id",
    name: "roles_edit",
    title: "Role edition",
    component: RoleForm,
    hidden: true,
    exact: true,
    strict: true
  },
  {
    path: "/new/roles",
    name: "roles_create",
    title: "New role",
    component: RoleForm,
    hidden: true,
    exact: true,
    strict: true
  },
  {
    path: "/business_units",
    name: "business_units",
    title: "Business Units",
    component: BusinessUnits,
  },
  {
    path: "/business_units/:id",
    name: "business_unit",
    title: "Business Unit details",
    component: BusinessUnitDetail,
    hidden: true,
    strict: true,
    exact: true
  },
  {
    path: "/new/business_units",
    name: "business_unit_new",
    title: "New Business Unit",
    component: BusinessUnitForm,
    hidden: true,
    strict: true,
    exact: true
  },
  {
    path: "/edit/business_units/:id",
    name: "business_unit_edit",
    title: "Edit Business Unit",
    component: BusinessUnitForm,
    hidden: true,
    strict: true,
    exact: true
  }
];
