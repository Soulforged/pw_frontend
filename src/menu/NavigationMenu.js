//@flow
import React from "react";
import { NavLink } from "react-router-dom";
import { images } from "src/resources";
import type { FeatureRoute } from "src/types";
import { compose, withHandlers } from "recompose";

type UserMenuIconProps = {
  menuCollapsed: boolean,
  userName: string,
  toggleUserMenu: (Event) => void,
  userMenuCollapsed: boolean
};

type UserMenuProps = {
  onSignOut: (Event) => void,
  show: boolean
};

type Props = {
  routes: FeatureRoute[],
  toggleMenu: (Event) => void,
  toggleTheme: (Event) => void,
  dark: boolean,
} & UserMenuIconProps & UserMenuProps;

const UserMenuIcon = (props: UserMenuIconProps) => {
  const {
    menuCollapsed,
    userName,
    userMenuCollapsed,
    toggleUserMenu
  } = props;
  return menuCollapsed ? (
    <div className="usrnm-small">
      <img alt="user-icon" src={images.userSmall} className="pointer" />
    </div>
  ) : (
    <div className="usrnm-big">
      <span>{userName}</span>
      <button className={`fa fa-chevron-${userMenuCollapsed ? "up" : "down"} pointer icon-button`} onClick={toggleUserMenu} />
    </div>
  );
};

const UserMenu = ({ show, onSignOut }: UserMenuProps) => (
  show ? (
    <div
      className="usrnm-div animate"
      style={show ? { opacity: 1 } : { opacity: 0 }}
    >
      <p>
        <a href="/#" className="change-pswd">Change Password</a>
      </p>
      <p>
        <a href="/#" className="sign-out" onClick={onSignOut}>Sign Out</a>
      </p>
    </div>
  ) : false
);

const Component = (props: Props) => (
  <div className="left-relative">
    <div className="left-up">
      <div className="text-right">
        <i className="menu-toggle fa fa-bars pointer" aria-hidden="true" onClick={props.toggleMenu} />
      </div>
      <div className={props.menuCollapsed ? "logo-small" : "logo-big"}>
        <img src={props.menuCollapsed ? images.logoSmall : images.logo} className="home pointer" alt="logo" />
      </div>
    </div>

    <div className="left-center">
      <div className="usrnm-lbl">
        <UserMenuIcon {...props} />
      </div>
      <UserMenu show={props.userMenuCollapsed !== false} onSignOut={props.onSignOut} />
    </div>

    <div className="left-down">
      <ul className="menu-nav">
        {props.routes && props.routes.filter(route => !route.hidden).map(route => (
          <li key={route.name} className="nav-tabs">
            <NavLink to={route.path}>
              <img src={images[route.name]} alt={route.name} />
              <span style={{ display: props.menuCollapsed ? "none" : "inline" }}>{route.title}</span>
            </NavLink>
          </li>
        ))}
      </ul>
    </div>

    <div className="absolute-down">
      <div className="theme_toggle">
        <label className="switch" htmlFor="theme_toggle">
          <input type="checkbox" id="theme_toggle" onChange={props.toggleTheme} checked={props.dark} />
          <span className="slider round" />
        </label>
      </div>
      <p className="help-support" style={{ display: props.menuCollapsed ? "none" : "block" }}>
        <a href="/">Help and support</a>
      </p>
    </div>
  </div>
);

export default compose(withHandlers({
  onSignOut: props => (event) => {
    event.preventDefault();
    props.logout();
  }
}))(Component);
