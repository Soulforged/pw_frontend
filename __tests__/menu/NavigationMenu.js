//@flow
import React from "react";
import { Router } from "react-router";
import NavigationMenu from "src/menu/NavigationMenu";
import renderer from "react-test-renderer";
import routes from "src/config/routes";
import createHistory from "history/createBrowserHistory";

it("matches snapshot no session", () => {
  const render = renderer.create(<NavigationMenu />).toJSON();
  expect(render).toMatchSnapshot();
});

it("matches snapshot with session", () => {
  const render = renderer.create((
    <Router history={createHistory()}>
      <NavigationMenu routes={routes} userName="testuser" />
    </Router>
  )).toJSON();
  expect(render).toMatchSnapshot();
});
