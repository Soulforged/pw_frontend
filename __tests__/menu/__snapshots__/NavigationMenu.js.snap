// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot no session 1`] = `
<div
  className="left-relative"
>
  <div
    className="left-up"
  >
    <div
      className="text-right"
    >
      <i
        aria-hidden="true"
        className="menu-toggle fa fa-bars pointer"
        onClick={undefined}
      />
    </div>
    <div
      className="logo-big"
    >
      <img
        alt="logo"
        className="home pointer"
        src="pegb-logo-final-white.svg"
      />
    </div>
  </div>
  <div
    className="left-center"
  >
    <div
      className="usrnm-lbl"
    >
      <div
        className="usrnm-big"
      >
        <span />
        <button
          className="fa fa-chevron-down pointer icon-button"
          onClick={undefined}
        />
      </div>
    </div>
    <div
      className="usrnm-div animate"
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <p>
        <a
          className="change-pswd"
          href="/#"
        >
          Change Password
        </a>
      </p>
      <p>
        <a
          className="sign-out"
          href="/#"
          onClick={[Function]}
        >
          Sign Out
        </a>
      </p>
    </div>
  </div>
  <div
    className="left-down"
  >
    <ul
      className="menu-nav"
    />
  </div>
  <div
    className="absolute-down"
  >
    <div
      className="theme_toggle"
    >
      <label
        className="switch"
        htmlFor="theme_toggle"
      >
        <input
          checked={undefined}
          id="theme_toggle"
          onChange={undefined}
          type="checkbox"
        />
        <span
          className="slider round"
        />
      </label>
    </div>
    <p
      className="help-support"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <a
        href="/"
      >
        Help and support
      </a>
    </p>
  </div>
</div>
`;

exports[`matches snapshot with session 1`] = `
<div
  className="left-relative"
>
  <div
    className="left-up"
  >
    <div
      className="text-right"
    >
      <i
        aria-hidden="true"
        className="menu-toggle fa fa-bars pointer"
        onClick={undefined}
      />
    </div>
    <div
      className="logo-big"
    >
      <img
        alt="logo"
        className="home pointer"
        src="pegb-logo-final-white.svg"
      />
    </div>
  </div>
  <div
    className="left-center"
  >
    <div
      className="usrnm-lbl"
    >
      <div
        className="usrnm-big"
      >
        <span>
          testuser
        </span>
        <button
          className="fa fa-chevron-down pointer icon-button"
          onClick={undefined}
        />
      </div>
    </div>
    <div
      className="usrnm-div animate"
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <p>
        <a
          className="change-pswd"
          href="/#"
        >
          Change Password
        </a>
      </p>
      <p>
        <a
          className="sign-out"
          href="/#"
          onClick={[Function]}
        >
          Sign Out
        </a>
      </p>
    </div>
  </div>
  <div
    className="left-down"
  >
    <ul
      className="menu-nav"
    >
      <li
        className="nav-tabs"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/dashboard"
          onClick={[Function]}
          style={undefined}
        >
          <img
            alt="dashboard"
            src="dashboard.svg"
          />
          <span
            style={
              Object {
                "display": "inline",
              }
            }
          >
            Dashboard
          </span>
        </a>
      </li>
      <li
        className="nav-tabs"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/users"
          onClick={[Function]}
          style={undefined}
        >
          <img
            alt="users"
            src="admin-icon.svg"
          />
          <span
            style={
              Object {
                "display": "inline",
              }
            }
          >
            Users
          </span>
        </a>
      </li>
      <li
        className="nav-tabs"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/roles"
          onClick={[Function]}
          style={undefined}
        >
          <img
            alt="roles"
            src="admin-icon.svg"
          />
          <span
            style={
              Object {
                "display": "inline",
              }
            }
          >
            roles
          </span>
        </a>
      </li>
      <li
        className="nav-tabs"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/business_units"
          onClick={[Function]}
          style={undefined}
        >
          <img
            alt="business_units"
            src="admin-icon.svg"
          />
          <span
            style={
              Object {
                "display": "inline",
              }
            }
          >
            Business Units
          </span>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="absolute-down"
  >
    <div
      className="theme_toggle"
    >
      <label
        className="switch"
        htmlFor="theme_toggle"
      >
        <input
          checked={undefined}
          id="theme_toggle"
          onChange={undefined}
          type="checkbox"
        />
        <span
          className="slider round"
        />
      </label>
    </div>
    <p
      className="help-support"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <a
        href="/"
      >
        Help and support
      </a>
    </p>
  </div>
</div>
`;
